src/System.IO.FileSystem/tests/FileStream/LockUnlock.cs

```diff
@@ -38,13 +38,27 @@ public void FileClosed_Throws()
             Assert.Throws<ObjectDisposedException>(() => fs.Lock(0, 1));
         }
 
-        [ActiveIssue(5964, TestPlatforms.OSX)]
+        [Fact]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        public void LockUnlock_Unsupported_OSX()
+        {
+            string path = GetTestFilePath();
+            File.WriteAllBytes(path, new byte[100]);
+
+            using (FileStream fs = File.Open(path, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite))
+            {
+                Assert.Throws<PlatformNotSupportedException>(() => fs.Lock(0, 100));
+                Assert.Throws<PlatformNotSupportedException>(() => fs.Unlock(0, 100));
```

Comment:
Might want to also verify that after these calls the FileShare'ing for the FileStream is still valid, i.e. that we didn't somehow mess up the lock that was already there.